src/hooks/useEvidences.js

```diff
@@ -30,11 +27,19 @@ export default function useEvidences(evidences) {
 
       // If is a valid CID try to fetch the content
       if (ipfsCid) {
+        // If the cid was already processed exit
+        if (evidenceCache.get(ipfsCid)) {
+          return
+        }
+
         const { data, error } = await ipfsGet(ipfsCid)
+
         if (error) {
           return appendEvidence({ error: ERROR_TYPES.ERROR_FETCHING_IPFS })
         }
         if (data) {
+          setEvidenceCache(evidenceCache.set(evidenceCache, true))
```

Comment:
I don't think you need to use the `setEvidenceCache()` here; the `Map` doesn't need to be re-set.